conda-store-server/conda_store_server/_internal/utils.py

```diff
@@ -147,3 +149,33 @@ def callable_or_value(v, *args, **kwargs):
     if callable(v):
         return v(*args, **kwargs)
     return v
+
+
+def compile_arn_sql_like(
+    arn: str, allowed_regex: re.Pattern[AnyStr]
+) -> tuple[str, str]:
+    """Turn an arn into a string suitable for use in a SQL LIKE statement.
+
+    Parameters
+    ----------
+    arn : str
+        String which matches namespaces and environments. For example:
+
+            */*          matches all environments
+            */team       matches all environments named 'team' in any namespace
+
+    allowed_regex : re.Pattern[AnyStr]
+        Regex to use to match the ARN.
+
+    Returns
+    -------
+    tuple[str, str]
+        (namespace regex, environment regex) to match in a sql LIKE statement.
+        See conda_store_server.server.auth.Authentication.filter_environments
+        for usage.
+    """
+    match = allowed_regex.match(arn)
+    if match is None:
+        raise ValueError(f"invalid arn={arn}")
```

Comment:
Let's make this a more user-friendly message. Otherwise it is ambiguous re wether the query was invalid/malformed or if there was no match.
Could even go as far as being `raise ValueError(f"Could not find a match for the requested namespace/environment={arn}")`
```suggestion
        raise ValueError(f"Could not find a match for the requested arn={arn}")
```

Reply:
Thanks, this is far better than the original.